### ISSUE TITLE
Pulling .rst function documentation out of .php

### ISSUE THREAD

**Issue by willhaynes** [INN/Largo#333]
#### What's currently working.

Using __[`doxphp`](https://github.com/avalanche123/doxphp)__ (a php phar), it's possible to generate a `.json` representation of documentation directly from our `.php` files.

From there, we can generate `*.rst` using the __`doxphp2sphinx`__ renderer supplied in the same phar.

	$ doxphp < functions.php | doxphp2sphinx > functions.rst

As a proof of concept, this is currently in our sphinx Makefile for a few select files [here](https://github.com/INN/Largo/blob/develop/docs/Makefile#L179-L185). With `make php`, It will generate an `.rst` files for each specified `.php` file. With `make html`, those files will compile to `.html`  [like this](http://largo.readthedocs.org/api/inc/helpers.html). It requires you to have **`doxphp`** installed.

**tl;dr:** `.php > .json > .rst > .html`

We could modify this pipeline, writing either our own `.php > .json` parser (seems unnecessary) or sphinx extension to render the included `.json > .rst` in a different way. In theory, this shouldn't be too hard.

#### Questions.

1.	Is organizing function documentation like this the best 	practice? (i.e is the way we organize our code is the best way to organize our function documentation?)

2.	Not all files have functions that need documentation. I'd assume most of what should be included resides in the `./inc/` folder, but is there anything else that should be included? Is there anything in `./inc/` that shouldn't be included?

#### Others.

What others are doing

* [WordPress Codex](http://codex.wordpress.org/): Organizes function reference one function per page. I tend to like this format and think our users would be familiar with it. It also would allow us to add examples and longer form documentation for those functions that need it.

* [WooThemes](http://docs.woothemes.com/): Seems to have their documentation all over the place.

	- Their **WooCommerce** plugin uses [APIgen](http://www.apigen.org/) to generate documentation for their WooCommerce plugin. and keeps it separate from 
	- **WooCodex** seems to mirror the structure of WordPress for some of their more [commonly used](http://docs.woothemes.com/documentation/woocodex/) functions ([like this](http://docs.woothemes.com/document/woocommerce_breadcrumb/)).

* Others?

**Comment by willhaynes** [comment]
@tothebeat can we do a mini audit of everything that's done here and what needs to be done next?

**Comment by rnagle** [comment]
Closing this since we have a working method of generating docs from PHP doc blocks and a couple hundred issues filed regarding adding and/or properly formatting doc blocks in each of the project's files.
